.github/workflows/build-check.yml

```
@@ -0,0 +1,25 @@
+name: Build Check
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
```

This can be bumped to v4, I think the Actions linter might complain about it being on V2

```suggestion
      uses: actions/checkout@v4
```